src/WebJobs.Script/Rpc/FunctionRegistration/FunctionRegistry.cs

```diff
@@ -66,7 +67,7 @@ public void Register(FunctionRegistrationContext context)
         public void WorkerError(WorkerErrorEvent workerError)
         {
             // TODO: move retry logic, possibly into worker channel decorator
-            _channelState.AddOrUpdate(workerError.Worker.Config,
+            _channelState.AddOrUpdate(_channelsDictionary[workerError.WorkerId].Config,
```

Comment:
If the channel is not found, that means there are no subscriptions attached to the channel. To avoid  null refs here, updated WorkerErrorEvent to enforce non null worker id and added a null check here.